### PR TITLE
Canada "county-level" data

### PR DESCRIPTION
**Pull request by jy2wong**

### CODE REVIEW

**Comment by beshaya** [comment]
Could you please post `update_prevalence.py` as a single PR so it is easier to review?

**Comment by jy2wong** [comment]
> Could you please post `update_prevalence.py` as a single PR so it is easier to review?

https://github.com/microcovid/microcovid/pull/904/files/01df49617ec7d92d207498167137b7df8aad223d?file-filters%5B%5D=.py

**Comment by jy2wong** [comment]
https://github.com/microcovid/microcovid/pull/904/files/1c26e53e8ce40231694d86d940ed57586fa14f41 is a link to a codebase diff containing all the update_prevalence.py changes together.

**Review by beshaya — APPROVED** [review]
LGTM % nits

**Comment by blanchardjeremy** [comment]
How do I test this? I see "sub sub location" in the code. But does that mean there is a third level of the location boxes that should be showing up?

File: src/components/calculator/PrevalenceControls.tsx [hunk 1]
```diff
@@ -155,6 +175,22 @@ export const PrevalenceControls: React.FunctionComponent<{
...
+    : Locations[data.subLocation].subdivisions.map((locKey) => {
+        return { label: Locations[locKey].label, value: locKey }
+      })
+  console.log(
+    'show non us counties:',
+    showSubSubLocation,
+    data.subLocation,
+    'options:',
+    subSubLocationOptions,
+  )
```
Comment (blanchardjeremy) [inline src/components/calculator/PrevalenceControls.tsx:189]: Do you still want this in the code when we commit?

**Review by blanchardjeremy — COMMENTED** [review]
Question

File: src/components/calculator/PrevalenceControls.tsx (hunk 1 above)
Comment (jy2wong) [reply src/components/calculator/PrevalenceControls.tsx:189]: nah.

**Comment by jy2wong** [comment]
> How do I test this? I see "sub sub location" in the code. But does that mean there is a third level of the location boxes that should be showing up?

Canada
Ontario
this should make the third box show up

File: update_prevalence.py [hunk 2]
```diff
@@ -481,19 +588,23 @@ class AppLocation(pydantic.BaseModel):
...
     # https://covid19-projections.com/estimating-true-infections-revisited/
-    def prevalanceRatio(self) -> float:
+    def prevalenceRatio(self) -> float:
         DAY_0 = datetime(2020, 2, 12)
         day_i = (datetime.now() - DAY_0).days
         positivityRate = self.positiveCasePercentage
         if positivityRate is None or positivityRate > 100:
             positivityRate = 100
+        if positivityRate < 0:
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (jy2wong) [reply update_prevalence.py:640]: done

**Pull request merged**